### ISSUE TITLE
Support Contextual Information in datapackage.json for WACZ

### ISSUE THREAD

**Issue by markpbaggett** [webrecorder/browsertrix-crawler#268]
With browsertrix-crawler, a user can use `combineWARC` to write contextual information defined in the `warcinfo` property into the destination warc. When the warc is read, the fields defined in the property can be read.

```
WARC/1.0
WARC-Filename: exhibitstwo_0.warc.gz
WARC-Date: 2023-03-29T13:58:06Z
WARC-Type: warcinfo
WARC-Record-ID: <urn:uuid:0b216a80-253c-4324-9dc7-a0e18e535b66>
Content-Type: application/warc-fields
Content-Length: 712

software: Browsertrix-Crawler 0.8.0-beta.2 (with warcio.js 1.6.2 pywb 2.7.3)
format: WARC File Format 1.0
title: Database of the Smokies
type: collection
operator: Mark Baggett askDI@utk.edu
hostname: www.lib.utk.edu
creator: University of Tennessee, Libraries
description: This warc contains web assets associated with the Database of the Smokies (DOTS). DOTS provided citations to written works about the Great Smoky Mountains National Park and bordering  communities from 1935 to the present day. In 2023, a decision was made to sunset DOTS but provide ongoing access to its contents. While DOTS will be spun down, its web archive will be made available for perpetuity.
```

There does not appear to be a way currently with Browsertrix-Crawler (0.8.0-beta.2) to do a similar thing where one could add contextual information about the contents of a WACZ to the datapackage.json file. The WACZ spec states that a "WACZ file includes all the data that is needed for the rendering archived content as well as [contextual information](https://specs.webrecorder.net/wacz/1.1.1/#dfn-context) required for users to interpret it." It goes further to state that "the datapackage.json SHOULD include properties that allow rendering applications to present the user with [contextual information](https://specs.webrecorder.net/wacz/1.1.1/#dfn-context) about the web archive:" including `title`, `description`, "other properties from the [[FRICTIONLESS-DATA-PACKAGE](https://specs.webrecorder.net/wacz/1.1.1/#bib-frictionless-data-package)] specification such as licenses, version, organization, contributors, email," and "custom properties that do not interfere with pre-existing properties."

Is there a way currently or are there plans to add a method to add contextual information from the config to `datapackage.json`? If not, I think this would be a great addition.

**Comment by ikreymer** [comment]
Yes, this is a good idea, and just something we haven't gotten around to yet. The warcinfo option (which I forgot about!) exists because someone requested it at one point.
I think its just a matter of exposing these settings.. are there specific properties that you'd like to see included? `title` and `description`?

A tricky aspect of this is that this sort of metadata may change while the actual archived data remains the same, eg. you may want to edit the `description`, but perhaps not. At least with WACZ, it is easier to update than in the WARC.

**Comment by ikreymer** [comment]
@tw4l what do you think would make sense here? Perhaps just adding a `--title` and `--desc` flags, or a more extensive metadata json blob in the config?

**Comment by tw4l** [comment]
@markpbaggett Initial PR is merged and the 0.9.0 release of the crawler will include `--title` and `--description` CLI args :)

Thanks for the full example of the metadata you'd like to be able to pass into `datapackage.json`! Keeping this issue open so that we can track adding more metadata to the WACZ in time.

**Comment by markpbaggett** [comment]
@tw4l and @ikreymer awesome 😎 .  this is a great start.
